File: filterx-matrix-20260820/projects/fastapi-sqlalchemy__angular/app/models/book.py
from sqlalchemy import Column, Float, ForeignKey, Integer, String
from sqlalchemy.orm import relationship
from app.database import Base

class Book(Base):
    __tablename__ = 'books'
    id = Column(Integer, primary_key=True)
    title = Column(String, nullable=False)
    genre = Column(String, nullable=False)
    price = Column(Float, nullable=False)
    note = Column(String, nullable=True)
    author_id = Column(Integer, ForeignKey('authors.id'), nullable=False)
    author = relationship('Author', back_populates='books')

File: filterx-matrix-20260820/projects/spring-boot-jpa__nextjs/src/main/java/com/example/model/Book.java
package com.example.model;
import jakarta.persistence.*;
import java.math.BigDecimal;
@Entity @Table(name="books") public class Book {
 @Id @GeneratedValue(strategy=GenerationType.IDENTITY) private Long id;
 @Column(nullable=false) private String title; @Column(nullable=false) private String genre;
 @Column(nullable=false) private BigDecimal price; private String note;
 @Column(name="author_id",nullable=false) private Long authorId;
 @ManyToOne(fetch=FetchType.LAZY) @JoinColumn(name="author_id",insertable=false,updatable=false) private Author author;
 public Long getId(){return id;} public String getTitle(){return title;} public String getGenre(){return genre;}
 public BigDecimal getPrice(){return price;} public String getNote(){return note;} public Long getAuthorId(){return authorId;} public Author getAuthor(){return author;}
}
